File: samples/authentication-as-app/ruby-oauth2/execute.rb
#!/usr/bin/env ruby

require "rubygems"
require "oauth2"

# Configuring credentials
credentials = File.read(File.expand_path("../../../credentials.txt")).split("\n");
keys = {
  :consumer_key => credentials[0],
  :consumer_secret => credentials[1]
}

client = OAuth2::Client.new(keys[:consumer_key], keys[:consumer_secret], 
                            :site => 'https://api.twitter.com',
                            :token_url => 'oauth2/token')
token = client.client_credentials.get_token

# The request to users/show doesn't depend
# on user context, so it's ok for auth as app
response = token.get('/1.1/users/show.json?screen_name=twitter')
puts response.body

File: samples/authentication-as-app/nodejs-simple-oauth2/execute.js
#!/usr/bin/env nodejs

var oauth = require('simple-oauth2');
var fs = require('fs');

// Configuring credentials
var credentials = fs.readFileSync(__dirname + '/../../../credentials.txt', 'utf8').split('\n');

var keys = {
  clientID: credentials[0],
  clientSecret: credentials[1],
  site: 'https://api.twitter.com',
  tokenPath: '/oauth2/token'
};

// Creating the client
var twitter = oauth(keys);

// Retrieve the token
twitter.client.getToken({}, function(err, result){
  if(err)
    throw err;

  var auth = twitter.accessToken.create(result);

  // Get user info
  twitter.api('GET', '/1.1/users/show.json?screen_name=twitter', {
    access_token: auth.token.access_token
  }, function(err, result){
    if(err)
      throw err;

    console.log(result);
  });
});
